python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -55,6 +56,14 @@
     "AutoscalerSummary",
     ["active_nodes", "pending_nodes", "pending_launches", "failed_nodes"])
 
+# Whether a worker should be kept based on the min_workers and
+# max_workers constraints.
```

Comment:
I think another constraint here is the ray.request_resources() which is similar to min_workers.

Reply:
Right now  ray.request_resources() is handled separately and does not affect KeepOrTerminate.
We might want to refactor it further, but I would do it in subsequent PR(s).